sdks/java/harness/src/test/java/org/apache/beam/fn/harness/BeamFnDataReadRunnerTest.java

```diff
@@ -178,7 +179,7 @@ public void testCreatingAndProcessingBeamFnDataReadRunner() throws Exception {
               null /* beamFnTimerClient */,
```

Comment:
bundleId is now unused is likely a linter issue, if its still used, please replace existing references to use DEFAULT_BUNDLE_ID in this test.

Reply:
https://github.com/apache/beam/pull/15011

I would not be oppose to adding additional locking to ProcessBundleHandler, but in general we try to avoid locks while invoking user code so this seems to imply we still need to have locking in BeamFnDataReadRunner (as well as it being good practice to guard against internal access of private variables that may be accessed from multiple threads).

Reply:
The guarantee would be that `reset()` won't be invoked while the other methods are being invoked.

We also need documentation saying which methods can be called in parallel with others.